hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/states/datanode/RunningDatanodeState.java

```diff
@@ -173,11 +118,32 @@ public void setExecutorCompletionService(ExecutorCompletionService e) {
 
   private Callable<EndPointStates> getEndPointTask(
       EndpointStateMachine endpoint) {
-    if (endpointTasks.containsKey(endpoint)) {
-      return endpointTasks.get(endpoint).get(endpoint.getState());
-    } else {
-      throw new IllegalArgumentException("Illegal endpoint: " + endpoint);
+    Callable<EndPointStates> endpointTask = null;
+    for (EndpointStateMachine endpointStateMachine : connectionManager.getValues()) {
+      if (endpointStateMachine.getAddressString().equals(endpoint.getAddressString())) {
```

Comment:
This for loop and the if-statement are not needed since `endpointStateMachine` is not used to build the task.